src/main.c

```diff
@@ -433,6 +490,67 @@ void WriteInterfaces()
     free(InterfaceArray);
 }
 
+void ParseRasNdisWanPacketFragment(PEVENT_RECORD ev)
+{
+    // Parse the current RasNdisWan packet event for use elsewhere.
+    // NB: Here we only do per-packet parsing. For any event fields that only need to be
+    // parsed once and written into an INTERFACE, we do the parsing in AddInterface.
+    ULONG Err;
+    wchar_t buffer[256];
+    PROPERTY_DATA_DESCRIPTOR Desc;
+    ULONG ParamNameSize = 0;
+    
+    // RoutingDomainID
+    Desc.PropertyName = (unsigned long long)(L"RoutingDomainID");
+    Desc.ArrayIndex = ULONG_MAX;
+    (void)TdhGetPropertySize(ev, 0, NULL, 1, &Desc, &ParamNameSize);
+    if ((ParamNameSize / sizeof(wchar_t)) != GUID_STR_SIZE)
+    {
+        printf("error decoding Microsoft-Windows-Ras-NdisWanPacketCapture: TdhGetPropertySize returns an unexpected RoutingDomainID size\n");
+        return;
+    }
+    Err = TdhGetProperty(ev, 0, NULL, 1, &Desc, sizeof(buffer), (PBYTE)buffer);
+    if (Err != NO_ERROR) {
+        buffer[0] = L'\0';
+    }
+    buffer[ParamNameSize / sizeof(wchar_t) + 1] = L'\0';
+    WideCharToMultiByte(CP_ACP,
+        0,
+        buffer,
+        -1,
+        RasNdisWanPacketFragment.RoutingDomainID,
+        ParamNameSize / sizeof(wchar_t) + 1,
+        NULL,
+        NULL);
+    RasNdisWanPacketFragment.RoutingDomainID[wcslen(buffer)] = '\0';
+
+    // Username
+    Desc.PropertyName = (unsigned long long)(L"Username");
+    Desc.ArrayIndex = ULONG_MAX;
+    (void)TdhGetPropertySize(ev, 0, NULL, 1, &Desc, &ParamNameSize);
+    Err = TdhGetProperty(ev, 0, NULL, 1, &Desc, sizeof(buffer), (PBYTE)buffer);
+    if (Err != NO_ERROR) {
+        buffer[0] = L'\0';
+    }
+    buffer[ParamNameSize / sizeof(wchar_t) + 1] = L'\0';
+    WideCharToMultiByte(CP_ACP,
+        0,
+        buffer,
+        -1,
+        RasNdisWanPacketFragment.Username,
+        ParamNameSize / sizeof(wchar_t) + 1,
+        NULL,
+        NULL);
+    RasNdisWanPacketFragment.Username[wcslen(buffer)] = '\0';
+
+    //Compute the ifPreHashValue - using the last part of the RoutingDomainID
+    if (strlen(RasNdisWanPacketFragment.RoutingDomainID) == GUID_STR_SIZE - 1) {
+        char lastPartOfGuid[8];
```

Comment:
Will there usually just be a single VPN interface? If so, then this point is moot; but assuming subsets of GUIDs are unique is pretty error-prone. So if there are actually a bunch of VPN interfaces, a better hash function should probably be used.

Also, why is this called a "PreHash"? Isn't it just a "Hash"?